### PR TITLE
add TS offline mode

### PR DESCRIPTION
**Pull request by Mercy811**
# Amplitude Developer Docs PR

## Description

Add offline mode section for the new feature 
image

Add config 
image

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [x] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sdks/browser-2/index.md [hunk 1]
```diff
@@ -420,3 +421,18 @@ amplitude.init("api-key", null, {
   identityStorage: "localStorage",
 });
 ```
+
+### Offline mode
+
+!!!note "Auto flush when reconnect"
+    Note that setting `config.flushIntervalMillis` to a very small value like 1 might cause `ERR_NETWORK_CHANGED`.
```
Comment (markzegarelli) [inline docs/data/sdks/browser-2/index.md:428]:
```suggestion
    Setting `config.flushIntervalMillis` to a small value like `1` may cause an `ERR_NETWORK_CHANGED` error.
```

File: docs/data/sdks/browser-2/index.md [hunk 2]
```diff
@@ -420,3 +421,18 @@ amplitude.init("api-key", null, {
...
+
+Starting from version 2.4.0, the Amplitude Browser SDK supports offline mode. The SDK checks network connectivity every time it tracks an event. If the device is connected to network, the SDK schedules a flush. If not, it saves the event to storage. The SDK also listens for changes in network connectivity and schedules a flush based on `config.flushIntervalMillis` for all stored events when the device reconnects.
```
Comment (markzegarelli) [inline docs/data/sdks/browser-2/index.md:430]:
```suggestion
Beginning with version 2.4.0, the Amplitude Browser SDK supports offline mode. The SDK checks network connectivity every time it tracks an event. If the device is connected to network, the SDK schedules a flush. If not, it saves the event to local storage. The SDK also listens for changes in network connectivity and schedules a flush of all stored events when the device reconnects, based on the `config.flushIntervalMillis` setting.
```

Is it accurate to specifically call out local storage here? I think we should be explicit that it's storing in the browser.

File: docs/data/sdks/browser-2/index.md [hunk 3]
```diff
@@ -420,3 +421,18 @@ amplitude.init("api-key", null, {
...
+
+To opt-out, refer to the code below. Otherwise, you can omit the configuration to keep it on.
```
Comment (markzegarelli) [inline docs/data/sdks/browser-2/index.md:432]:
```suggestion
To disable offline mode, add `offline: amplitude.Types.OfflineDisabled` to the `amplitude.init()` call as shown below.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

**Review by justin-fiedler — APPROVED** [review]
Thanks @Mercy811 !

File: docs/data/sdks/browser-2/index.md (hunk 2 above)
Comment (Mercy811) [reply docs/data/sdks/browser-2/index.md:430]: No, local storage is the default option. But customers can use their custom storages too. So I'd keep it without storage here. But thanks for calling out!

Comment (Mercy811) [reply docs/data/sdks/browser-2/index.md]: Updated with the rest of the suggest at https://github.com/amplitude/amplitude-dev-center/pull/159/commits/f0383800dd435502b9cc69578f410315229ce701 and https://github.com/amplitude/amplitude-dev-center/pull/159/commits/d4106d8dd7cadcf314c05675dd6172fd3d8b1d0a

**Review by markzegarelli — APPROVED** [review]

**Pull request merged**